source/SH3/graphics/texture.cpp

```diff
@@ -9,52 +9,380 @@
+
+        palette.resize(colorsPerBlock * nBlocks);
+
+        for(std::size_t block = 0; block < nBlocks; ++block)
+        {
+            std::size_t read = file.ReadData(static_cast<void*>(&palette[block * colorsPerBlock]), pal_header.entrySize, e);
+
+            if(read != pal_header.entrySize)
+            {
+                Log(LogLevel::WARN, "sh3_texture::Load( ): Warning: Number of bytes read in palette block != pal_header.entrySize! (Expected %d, got %d)", read, pal_header.entrySize);
+            }
+
+            ASSERT(read == pal_header.entrySize);
+            // Each block is 256 bytes.
+            // Seek to next block.
+            file.Seek(256 - pal_header.entrySize, std::ios_base::cur);
+        }
+
+        // we need to swap some colors every 32 pixels, starting from the 8th
+        static constexpr std::size_t swapDistance = 32; // distance between swaps
+        static constexpr std::size_t swapSize = 8; // amount of colors to swap
+        ASSERT(palette.size() > 8);
+
+        if(palette.size() > 8)
+        {
+            for(auto iter = next(begin(palette), 8); distance(iter, end(palette)) > swapDistance; advance(iter, swapDistance))
+            {
+                // swap 8 colors
+                const auto swapBlock = next(iter, swapSize);
+                if(distance(swapBlock, end(palette)) < swapSize)
+                {
+                    Log(LogLevel::WARN, "Palette doesn't have enough colors left for swapping.");
+                    break;
+                }
+                std::swap_ranges(iter, swapBlock, swapBlock);
+            }
+        }
+
+
+        // Now that we've completely loaded the palette in its entirety, we can get the 8-bit index value
+        // from the data section of the file and get it's color in the palette!
+
+        //===---THIS IS A CLUSTER FUCK FOR NOW UNTIL WE UNDERSTAND HOW IN THE NAME OF CHRIST THIS WORKS---===//
+        //std::vector<std::uint8_t> iBuffer;  // Our index buffer that we put transformed indecies into
+        std::vector<std::uint8_t> iBuffer;  // Our index buffer that we put transformed indecies into
+
+        data.resize(header.texWidth * header.texHeight * 3); // We strip the Alpha channel from the BGRA pixel beacuse it is hard locked to 0x80 (not 0xFF!!)
+        iBuffer.resize(header.texSize);
+
+        file.Seek(offset + (header.texFileSize - header.texSize), std::ios_base::beg); // Seek to the beginning of data
+
+        if(header.texWidth > 96) // Apparently this is the distortion flag?!?!
+        {
+            ASSERT(header.texWidth % 16 == 0);
+
+            if(header.texWidth % 16 != 0)
+            {
+
+            }
+
+            std::uint32_t x = 0;
+            std::uint32_t y = 0;
+            bool offsetFlipper = false;
+
+            // We need to step through each index and unencrypt it
+            while(true) // FIXME: distortion on 16-pixel wide block on the left
```

Comment:
I think this makes it look like the comment belongs to the looping condition, while (heh) it refers to the whole block.
I'd put it in a separate line above.